scipy/sparse/_index.py

```diff
@@ -31,18 +31,28 @@ class IndexMixin:
     """
     def __getitem__(self, key):
         row, col = self._validate_indices(key)
+
+        def vector(data):
+            if self._is_array:
+                raise NotImplementedError(
+                    'We have not yet implemented 1D sparse slices; '
+                    'please index using explicit indices, e.g. `x[:, [0]]`'
+                )
+            else:
+                return data
+
         # Dispatch to specialized methods.
         if isinstance(row, INT_TYPES):
             if isinstance(col, INT_TYPES):
                 return self._get_intXint(row, col)
             elif isinstance(col, slice):
-                return self._get_intXslice(row, col)
+                return vector(self._get_intXslice(row, col))
             elif col.ndim == 1:
                 return self._get_intXarray(row, col)
```

Comment:
This is also a 1d slice and needs to be checked.

Reply:
Also in need of checks: both calls to `_get_arrayXint` below.

Reply:
Thanks, this helped me catch some bugs---including one that was already there!

Reply:
Addressed.